### ISSUE TITLE
graphics() and bitimage() not working on Okipos 80 Plus III

### ISSUE THREAD

**Issue by mikersee** [mike42/escpos-php#7]
My printer, according to the manual is ESC/POS compatible, and the library works well with this and other printers I have, however on this particular printer, It can't print images, sending an image to either of my OkiPos 80 Plus III printers results in the printer printing some garbled text, and then freezing, (I have to unplug it) I know it can print images, because I connected it to a computer running Windows 95, and it can print images; but perhaps it can't print images in the ESC/POS way?

Also note the library works very well with the following printers: ( I will add more as I can)
Epson:
Epson TM-T88III (graphics() yes,  bitimage() yes, character test works too)
Okipos (Winbond/Nuvotron chip):
Okipos 80 Plus III (parallel interface, (graphics() no,  bitimage() no, character test works too))
Zijiang (Chinese company, Winbond/Nuvotron used)
ZJ-5890T (USB interface , graphics() no, bitimage () yes)
ZJ-5870 (USB interface , graphics() no, bitimage () yes)
NT-58H (USB interface , graphics() no, bitimage () yes)
Xprinter (another Chinese company, higher quality than ZiJiang) (testing these in next couple days)
XP-T58K
XP-58III
XP-58IIIA
XP-58IIIK
XP-Q800 (80mm auto cutter) all functions apart from graphics works
XP-76IIN (label printer, 58mm)

**Comment by estuaco** [comment]
Hi,

I tested in Seiko Epson Corp. Receipt Printer M129C/TM-T70 compatible, and bitImage and graphics not work.

**Comment by mikersee** [comment]
My Epson M129C , Bitimage and graphics are both working

**Comment by mike42** [comment]
Hi mikersee, thanks for the feedback. It's good to see what printers this is being used on "in the wild".

I've dug up some old documentation on Google, and it looks like the OkiPos 80 Plus III supports a different variant of the bitImage command to the one I've implemented. I'm currently working on supporting international character sets, but once this is done, I may be able to add some compatibility flags for it. Printing is quite complex, and there are 6 or so more print functions, plus 2 more output formats which I'm hoping to implement.

Hi estuaco, sorry to hear it's not working. I've got some questions. If you can answer these in a separate issue, I'd be happy to take a look.
- Which model is your printer? Is it actually Epson or "Epson-compatible"? (Model number on its label would be ideal)
- What happens when you try to print? Any PHP errors? (you need the GD library in PHP for the image functions)
- How are you trying to print? USB/Ethernet/Serial? Do the other test scripts produce formatted output? Are you testing with bit-image.php and graphics.php with the included sample images? (only PNG is supported at the moment)

**Comment by mikersee** [comment]
Is there any POS software using this PHP Esc/pos? I really need a good PHP pos software

**Comment by mike42** [comment]
Not that I know of. This was developed to add drop-in support for receipt printing to any PHP app, but it is BYO app.

The usual approach I've seen in web-based POS systems is to present a receipt in a small window as a web page, and use the local computer's driver to print it (which is not easy to adapt to the direct server-side printing that this library supports)

**Comment by mike42** [comment]
Closing this one, the Okipos 80 Plus III image issue is now covered by bug #83
